src/sentry/services/hybrid_cloud/audit/impl.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from typing import Any, Mapping
+
+from sentry.models import AuditLogEntry, AuthIdentity, AuthProvider, Organization, Team, User
+from sentry.services.hybrid_cloud.audit import AuditLogMetadata, AuditLogService
+from sentry.services.hybrid_cloud.auth import ApiAuthIdentity, ApiAuthProvider
+from sentry.services.hybrid_cloud.organization import ApiOrganizationMember
+from sentry.services.hybrid_cloud.user import APIUser, user_service
+
+
+class DatabaseBackedAuditLogService(AuditLogService):
```

Comment:
A high level concern: right now, I actually already did implement an AuditLogEntry service:

```
    def save_or_write_to_kafka(self):
        """
        Region Silos do not have access to the AuditLogEntry table which is specific to the control silo.
        For those silos, this method publishes the attempted audit log write to a durable kafka queue synchronously
        that will eventually be consumed by the control silo.  For the control silo, this method ultimately results
        in a save() call.
        """
        from sentry.region_to_control.producer import audit_log_entry_service

        audit_log_entry_service.produce_audit_log_entry(self)
```

Which is hooked up to some existing call sites, but isn't running in production.  However, it is currently used in a handful of tests.

One major difference is that in my original design, AuditLogs were written to kafka and asynchronously written to the database.  I do think we want to move *away from kafka*, but I wonder if using Outbox to synchronize these might also be a strong option.  It would improve latency and availability of these code paths since writing an AuditLog is never a synchronously required action.

Strictly speaking, such work is not necessary for this PR for now (let's get the auth login work across the line), but when the login work is wrapping up, let's think about how we can converge on a single audit log service, and wether we think it should remain synchronous or not.